### PR TITLE
Add HandlerIntegrationTest

### PR DESCRIPTION
**Pull request by shiliyu**
1. Remove `Resource not found` to more meaningful `statusCode` name.
2. Test fixed `special case when invokeFinalizingHandler with null`.
3. Test handler execution behaviors.

### CODE REVIEW

File: core/src/test/java/io/forestframework/core/http/routing/HandlerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    @CsvSource({
+            "errorInPreHandler",
+            "errorInHandler"
+    })
+    void exceptionsInPrehandlerAndHandlerAreCapturedByCustom500Handler(String handler) throws IOException {
+        String path = "/custom500/" + handler;
+        String result = sendHttpRequest("GET", path).assert200().getBody();
+
+        System.out.println("result: " + result);
+        System.out.println(this.getTraces());
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/HandlerIntegrationTest.java]:
```suggestion
```

File: core/src/test/java/io/forestframework/core/http/routing/HandlerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        assertThat(result, not(containsString("Resource not found")));
+    }
+
+    @Test
+    void error404IsHandledByCustom404ErrorHandler() throws IOException {
+        String path = "/custom4XXerror/inexistence";
+        String result = sendHttpRequest("GET", path).assert200().getBody();
+
+        System.out.println(getTraces());
+        System.out.println(result);
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/HandlerIntegrationTest.java]:
```suggestion
```

File: core/src/test/java/io/forestframework/core/http/routing/HandlerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+                        "prehandlerReturnsString",
+                        Message.CUSTOM_ERROR_HANDLER.name(),
+                        "shouldContinueInPostHandler"
+                ), getTraces());
+
+        assertThat(getTraces(), not(hasItem(Message.HANDLER.name())));
+        assertThat(result, containsString("java.lang.ClassCastException"));
+
+        assertThat(result, not(containsString("Resource not found")));
+    }
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/HandlerIntegrationTest.java]: We should throw errors at startup time, not runtime. I.e. inspect `Routing.getHandlerMethod()` return type via reflection, if the return type is not `void`/`boolean`/`Boolean`/`Future<Void>`/`CompletableFuture<Void>`/`Future<Boolean>`/`CompletableFuture<Boolean>`, throw exception and abort.

Comment (shiliyu) [reply core/src/test/java/io/forestframework/core/http/routing/HandlerIntegrationTest.java]: ‘kay, will fix this.

**Pull request closed without merging**